### DISCUSSION TITLE
Block websites

### DISCUSSION THREAD

**Discussion by brianjohnbooth** [ttscoff/bunch-docs#131] [Q&A] [answered]
Is there a way to block certain website domains? Or to allow access to only certain websites during a bunch?

**Comment by ttscoff** [comment, marked as answer]
No, bunch does nothing to restrict usage or filter network traffic. There are plenty of apps for that :).

- Brett

**Comment by ttscoff** [comment]
To further explain: A Bunch is either open or closed, but unless the Bunch is running a user-defined script that does otherwise, it's not monitoring or affecting the system in any way. Blocking access to a URL would require monitoring network traffic, which is outside Bunch's scope and more invasive than Bunch intends to be.

There are some apps that look really good for this purpose, such as [Session](https://setapp.com/apps/session) and [Focus](https://heyfocus.com/) (both on Setapp), but they don't seem to have external AppleScript automation. Which is weird because they both have internal methods for running scripts when starting/stopping. Buuut, you could probably do some AppleScript UI automation to turn them on/off.

You can also do [UI scripting with SelfControl](https://hints.macworld.com/article.php?story=20100801214648362), an older app that's a bit more strict than I would personally want (like once you start it there's no way to stop it until timer runs out). 

If anyone has a good suggestion for a _scriptable_ web content filter, please add it here!
